### DISCUSSION TITLE
Lesson 4 :Errors while installing web3 (MacOS) (SOLVED)

### DISCUSSION THREAD

**Discussion by JamiDEV** [smartcontractkit/full-blockchain-solidity-course-py#233] [Q&A] [answered]
Not sure where to begin as I'm still pretty confused as to what's going on.
![Screen Shot 2021-10-20 at 4 26 12 PM](https://user-images.githubusercontent.com/53233466/138181370-b744b66e-56d5-4a04-a11b-958a6db9d54e.png)

this is the second error at the bottom list of red

![Screen Shot 2021-10-20 at 4 26 58 PM](https://user-images.githubusercontent.com/53233466/138181452-70f245aa-501d-4bec-ad0c-bb2524236f3b.png)

**Comment by cromewar** [comment, marked as answer]
It seems the problem is you don't have a `C` compiler on your machine, the compiler you should use is gcc which actually gets installed with xcode.

Try to follow this instructions: https://mkyong.com/mac/how-to-install-gcc-compiler-on-mac-os-x/

**Reply by JamiDEV** [reply]
I'll give it a try thank you

**Reply by PatrickAlphaC** [reply]
Can you please post the text of the error and not the image?

**Reply by JamiDEV** [reply]
yes, I can do that for the next error I experience, my apologies.

**Comment by JamiDEV** [comment]
I did have to install a slew of things for a simple fix, however, Xcode command-line tools and the web3 virtual environments were recommended installs anyway. 


the Cython package was the last, I think that's what was missing. look at the top image, working from the bottom upward, count 6 lines and you'll see it says Cython not installed. lol, that was it. for me.

**Reply by cromewar** [reply]
Yeah that's why I recommended installing gcc because cython needs it as compiler, did you solve the problem tough?

**Reply by JamiDEV** [reply]
Yes, after the GCC install and then a Cython install I was able to resolve this issue.

**Reply by PatrickAlphaC** [reply]
Perfect! Can you post the link you used to do this?

**Reply by JamiDEV** [reply]
As cromewar mentioned earlier, you must have an Apple ID to get dev privileges, and then you can follow the link he provided.

[https://mkyong.com/mac/how-to-install-gcc-compiler-on-mac-os-x/](url)

You can then use Cython. org documentation to install cython.
[https://docs.cython.org/en/latest/src/quickstart/install.html](url)

**Reply by PatrickAlphaC** [reply]
boom!
